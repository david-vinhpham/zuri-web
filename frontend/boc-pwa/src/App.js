import React from 'react';
import Loadable from 'react-loadable';
import logo from './logo.svg';
import './App.css';

const AsyncComponent = Loadable({
  loader: () => import('./components/Loading'),
  loading: () => (<div>this is loadable loading</div>),
});

function App() {
  return (
    <div className="App">
      {/*<header className="App-header">*/}
      {/*  <img src={logo} className="App-logo" alt="logo" />*/}
      {/*  <p>*/}
      {/*    Edit <code>src/App.js</code> and save to reload.*/}
      {/*  </p>*/}
      {/*  <a*/}
      {/*    className="App-link"*/}
      {/*    href="https://reactjs.org"*/}
      {/*    target="_blank"*/}
      {/*    rel="noopener noreferrer"*/}
      {/*  >*/}
      {/*    Learn React*/}
      {/*  </a>*/}
      {/*</header>*/}
      <AsyncComponent />
    </div>
  );
}

export default App;
