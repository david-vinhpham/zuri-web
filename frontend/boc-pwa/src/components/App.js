import React from 'react';
import PropTypes from 'prop-types';
import Error from 'components/Error';
import Loading from 'components/Loading';
import { Provider as ReduxProvider } from 'react-redux';

const ContextType = {
  // Enables critical path CSS rendering
  // https://github.com/BOCVN/isomorphic-style-loader
  insertCss: PropTypes.func.isRequired,
  // Universal HTTP client
  fetch: PropTypes.func.isRequired,
  pathname: PropTypes.string.isRequired,
  query: PropTypes.object,
  // Integrate Redux
  // http://redux.js.org/docs/basics/UsageWithReact.html
  ...ReduxProvider.childContextTypes,
  // ReactIntl
  locale: PropTypes.string,
  store: PropTypes.object.isRequired,
};

class App extends React.PureComponent {
  static propTypes = {
    context: PropTypes.shape(ContextType).isRequired,
    children: PropTypes.element.isRequired,
  };
  static childContextTypes = ContextType;

  getChildContext() {
    return this.props.context;
  }

  render() {
    const { context, children } = this.props;
    const { store } = context;

    return (
      <ReduxProvider store={store}>
        <>
          <Error />
          <Loading />
          {React.Children.only(children)}
        </>
      </ReduxProvider>
    );
  }
}

export default App;
