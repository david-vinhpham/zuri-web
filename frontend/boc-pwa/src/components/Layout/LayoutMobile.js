/**
 * BOC VN (http://www.bocvietnam.com/)
 *
 * Copyright Â© 2018-present BOCVN, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import classnames from 'classnames';
import MessageNotSupport from './MessageNotSupport';

import s from './Layout.css';
import Header from '../Header';
import Footer from '../Footer';

class Layout extends React.Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    fullHeight: PropTypes.bool,
    fullWidth: PropTypes.bool,
  };

  static defaultProps = {
    fullHeight: false,
    fullWidth: false,
  };

  /**
   * This layout for mobile only
   * If desktop enter, got MessageNotSupport
   */
  render() {
    const { fullHeight, fullWidth } = this.props;
    return (
      <div
        className={classnames(s.layout, {
          [s.fullHeight]: fullHeight,
          [s.fullWidth]: fullWidth,
        })}
      >
        <Header hiddenXs />
        <div className={s.mobile}>{this.props.children}</div>
        <div className={s.desktop}>
          <MessageNotSupport />
        </div>
        <Footer hiddenXs />
      </div>
    );
  }
}

export default withStyles(s)(Layout);
