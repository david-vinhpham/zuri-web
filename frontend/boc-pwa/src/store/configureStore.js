import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import { persistStore, persistReducer } from 'redux-persist';

import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';

import storage from 'redux-persist/lib/storage';
import { name, version } from '../../package.json';
import rootReducer from '../reducers';
import createLogger from './logger';

export default function configureStore(initialState, helpersConfig) {
  const middleware = [thunk.withExtraArgument(helpersConfig)];

  const persistConfig = {
    key: 'BOCVN',
    storage,
    stateReconciler: autoMergeLevel2,
    blacklist: ['clz'],
  };

  const persistedReducer = persistReducer(persistConfig, rootReducer);

  let enhancer;

  if (__DEV__) {
    middleware.push(createLogger());

    // https://github.com/zalmoxisus/redux-devtools-extension#14-using-in-production
    const composeEnhancers = composeWithDevTools({
      // Options: https://github.com/zalmoxisus/redux-devtools-extension/blob/master/docs/API/Arguments.md#options
      name: `${name}@${version}`,
    });

    // https://redux.js.org/docs/api/applyMiddleware.html
    enhancer = composeEnhancers(applyMiddleware(...middleware));
  } else {
    enhancer = applyMiddleware(...middleware);
  }

  // https://redux.js.org/docs/api/createStore.html
  const store = createStore(persistedReducer, enhancer);

  // Hot reload reducers (requires Webpack or Browserify HMR to be enabled)
  if (__DEV__ && module.hot) {
    module.hot.accept('../reducers', () =>
      // Don't forget to remove `()` if you change reducers back to normal rootReducer.
      // eslint-disable-next-line global-require
      store.replaceReducer(require('../reducers').default()),
    );
  }

  const persistor = persistStore(store);

  return { store, persistor };
}
