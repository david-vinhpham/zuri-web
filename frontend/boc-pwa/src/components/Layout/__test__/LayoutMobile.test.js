/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import App from '../../App';
import LayoutMobile from '../LayoutMobile';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initialState = {
  runtime: {
    availableLocales: ['en-US'],
  },
  intl: {
    locale: 'en-US',
  },
};

describe('LayoutMobile', () => {
  test('renders children correctly', () => {
    const store = mockStore(initialState);

    const wrapper = renderer
      .create(
        <App
          context={{
            insertCss: () => {},
            fetch: () => {},
            pathname: '',
            store,
          }}
        >
          <LayoutMobile>
            <div className="child" />
          </LayoutMobile>
        </App>,
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });

  test('renders mobile layout with fullHeight and fullWidth', () => {
    const store = mockStore(initialState);
    const wrapper = renderer
      .create(
        <App
          context={{
            insertCss: () => {},
            fetch: () => {},
            pathname: '',
            store,
          }}
        >
          <LayoutMobile fullHeight fullWidth>
            <div className="child" />
          </LayoutMobile>
        </App>,
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
