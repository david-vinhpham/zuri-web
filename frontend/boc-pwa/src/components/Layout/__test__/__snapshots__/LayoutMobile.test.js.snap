// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LayoutMobile renders children correctly 1`] = `
<div
  className="layout"
>
  <div
    className="root hiddenXs"
  >
    <div
      className="logo"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="Homecredit"
          height="40"
          src="logo-inverse.png"
        />
      </a>
    </div>
  </div>
  <div
    className="mobile"
  >
    <div
      className="child"
    />
  </div>
  <div
    className="desktop"
  >
    <div
      className="notSupportMessage"
    >
      <img
        alt="sad-ico.svg"
        className="notSupportImg"
        src="sad-ico.svg"
      />
      <p>
        Hiện nay đường dẫn chỉ hỗ trợ các thiết bị điện thoại di động. Xin lỗi Quý khách vì sự bất tiện này.
      </p>
    </div>
  </div>
  <div
    className="root hiddenXs"
  >
    <div
      className="container"
    >
      <span
        className="text"
      >
        © 2018 Bản quyền thuộc về Home Credit
      </span>
    </div>
  </div>
</div>
`;

exports[`LayoutMobile renders mobile layout with fullHeight and fullWidth 1`] = `
<div
  className="layout fullHeight fullWidth"
>
  <div
    className="root hiddenXs"
  >
    <div
      className="logo"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="Homecredit"
          height="40"
          src="logo-inverse.png"
        />
      </a>
    </div>
  </div>
  <div
    className="mobile"
  >
    <div
      className="child"
    />
  </div>
  <div
    className="desktop"
  >
    <div
      className="notSupportMessage"
    >
      <img
        alt="sad-ico.svg"
        className="notSupportImg"
        src="sad-ico.svg"
      />
      <p>
        Hiện nay đường dẫn chỉ hỗ trợ các thiết bị điện thoại di động. Xin lỗi Quý khách vì sự bất tiện này.
      </p>
    </div>
  </div>
  <div
    className="root hiddenXs"
  >
    <div
      className="container"
    >
      <span
        className="text"
      >
        © 2018 Bản quyền thuộc về Home Credit
      </span>
    </div>
  </div>
</div>
`;
